src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -150,6 +238,229 @@ def get_S3FileSystem_from_session(
     return s3_fs
 
 
+def get_export_end_date(filename: str) -> Union[str, None]:
+    """Gets the export end date (converted to isoformat) based on the filename.
+
+        We can have filenames with with:
+        - {DateType}_{YYYYMMDD}-{YYYYMMDD}.json
+        - {DateType}_{YYYYMMDD}.json format
+        This function does handling for both. The export end date is always the
+        last component of the filename.
+
+    Args:
+        filename (str): name of the input json file
+
+    Returns:
+        Union[str, None]: export end date in isoformat if it exists
+    """
+    filename_components = os.path.splitext(filename)[0].split("_")
+    if len(filename_components) <= 1:
+        export_end_date = None
+    else:
+        if "-" in filename_components[-1]:
+            _, end_date = filename_components[-1].split("-")
+            end_date = datetime.datetime.strptime(end_date, "%Y%m%d")
+        else:
+            end_date = datetime.datetime.strptime(filename_components[-1], "%Y%m%d")
+        export_end_date = end_date.isoformat()
+    return export_end_date
```

Comment:
```suggestion
    if (
        filename is None
        or filename == ""
        or len((filename_components := os.path.splitext(filename)[0].split("_"))) <= 1
    ):
        return None

    if "-" in filename_components[-1]:
        _, end_date = filename_components[-1].split("-")
        end_date = datetime.datetime.strptime(end_date, "%Y%m%d")
    else:
        end_date = datetime.datetime.strptime(filename_components[-1], "%Y%m%d")
    export_end_date = end_date.isoformat()
    return export_end_date
```

Reply:
I like to return early in the invalid cases that way the rest of the function doesn't need to be nested within the else.